internal/prober/prober_test.go

```diff
@@ -33,192 +42,159 @@ import (
 	"github.com/go-logr/logr"
 	"github.com/golang/mock/gomock"
 	. "github.com/onsi/gomega"
-
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/version"
-	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 var (
-	config                              *papi.Config
-	ctrl                                *gomock.Controller
-	mds                                 *mockscaler.MockScaler
-	msc                                 *mockprober.MockShootClientCreator
-	clientBuilder                       *fake.ClientBuilder
-	fakeClient                          client.WithWatch
-	mki                                 *mockinterface.MockInterface
-	mdi                                 *mockdiscovery.MockDiscoveryInterface
-	errNotIgnorable                     = errors.New("not Ignorable error")
-	internalProbeFailureBackoffDuration = metav1.Duration{Duration: time.Millisecond}
-	proberTestLogger                    = logr.Discard()
-	defaultProbeTimeout                 = metav1.Duration{Duration: 40 * time.Second}
+	ctrl                        *gomock.Controller
+	mockScaler                  *mockscaler.MockScaler
+	mockShootClientCreator      *mockprober.MockShootClientCreator
+	mockKubernetesInterface     *mockinterface.MockInterface
+	mockDiscoveryInterface      *mockdiscovery.MockDiscoveryInterface
+	mockCoordinationV1Interface *mockcoordinationv1.MockCoordinationV1Interface
+	mockLeaseInterface          *mockcoordinationv1.MockLeaseInterface
+	unknownError                = errors.New("unknown error")
+	proberTestLogger            = logr.Discard()
+	testProbeTimeout            = metav1.Duration{Duration: 3 * time.Millisecond}
+	testProbeInterval           = metav1.Duration{Duration: 3 * time.Millisecond}
+	nodeLease1FilePath          = filepath.Join(testdataPath, "node_lease_1.yaml")
+	nodeLease2FilePath          = filepath.Join(testdataPath, "node_lease_2.yaml")
 )
 
-type probeStatusEntry struct {
-	name                              string
-	err                               error
-	expectedInternalProbeSuccessCount int
-	expectedInternalProbeErrorCount   int
-	expectedExternalProbeSuccessCount int
-	expectedExternalProbeErrorCount   int
+type probeSimulatorEntry struct {
```

Comment:
it is actually a test case struct entry. Therefore a more appropriate name would be `probeTestCase. Then in the tests you can say:
```go
testcases := []probeTestCase{}
```
But i find this approach a little sub-optimal for contributors. Reason is that for test case `TestAPIServerProbeFailure` you do not need most of the fields of the struct. Would it not be better to define the struct inline for every test case so that only what is required has to be set?